### PR TITLE
feat(data-pipeline): add concurrent idempotent processing flow

### PR DESCRIPTION
**Pull request by Irahan2**
#19 
- Replace sequential page loop with bounded parallel batches in Prefect flow

- Add hash-based claim/processed/failed tracking in Neo4j with stale claim reclaim

- Add race-safe schema reflection strategy (before/after batches) and run summary counters

- Add integration-style tests for 10-page processing, duplicate skip, and failure isolation

- Document DATA_PIPELINE_MAX_CONCURRENCY and DATA_PIPELINE_CLAIM_STALE_MINUTES

### CODE REVIEW

File: src/data_pipeline/flows/graph_populating.py [hunk 1]
```diff
@@ -17,6 +28,103 @@ def __init__(self):
...
+            REMOVE doc.claim_token
+            """,
+            params={
+                "doc_hash": doc_hash,
+                "error_message": error_message[:1000],
+            },
+        )
+
     def execute_cypher(self, query: str):
         logger = get_run_logger()
```
Comment (jgilewicz) [inline src/data_pipeline/flows/graph_populating.py]: get_run_logger() might throw an exception outside prefect pipeline, it might be better to use standard logger

File: src/data_pipeline/pipeline.py [hunk 2]
```diff
@@ -32,24 +66,116 @@ def data_pipeline_flow():
...
+
+        for page_index, page_content in enumerate(
+            pages[batch_start:batch_end],
+            start=batch_start + 1,
+        ):
+            page_hash = _compute_page_hash(page_content)
+            claim_future = claim_document_for_processing.submit(page_hash)
+            batch_claims.append((page_index, page_content, page_hash, claim_future))
+
+        for page_index, page_content, page_hash, claim_future in batch_claims:
```
Comment (jgilewicz) [inline src/data_pipeline/pipeline.py]: The problem in this loop is that it when claim for page 3 is ready, but page 2 is slow - 3 will not start. Its better to gather all of claims first and then submit like
```
for ..., claim_future in batch_claims:
    is_claimed = claim_future.result()
```

I don't really know if that would work, but its worth to test it out. If not - ack comment

File: src/data_pipeline/flows/graph_populating.py (hunk 1 above)
Comment (Irahan2) [reply src/data_pipeline/flows/graph_populating.py]: To be honest yes ,great catch. I added a safe logger helper that uses Prefect run logger when available and falls back to standard module logger outside Prefect context. Updated in graph_populating flow.

File: src/data_pipeline/pipeline.py (hunk 2 above)
Comment (Irahan2) [reply src/data_pipeline/pipeline.py]: I updated claim handling to process futures with as_completed, so a slow claim no longer blocks later ready claims in the batch. Also re-ran concurrency tests after the change. Let's hope that everything will work correctly xDD
